activerecord/lib/active_record/railtie.rb

```diff
@@ -214,8 +224,15 @@ class Railtie < Rails::Railtie # :nodoc:
         configs.each do |k, v|
           next if k == :encryption
           setter = "#{k}="
-          next if ActiveRecord.respond_to?(setter)
-          send(setter, v)
+          # Some existing initializers might rely on Active Record configuration
+          # being copied from the config object to their actual destination when
+          # `ActiveRecord::Base` is loaded.
+          # So to preserve backward compatibility we copy the config a second time.
+          if ActiveRecord.respond_to?(setter)
+            ActiveRecord.send(setter, v)
```

Comment:
Do we still need copying the config a first time in L212-L237?

https://github.com/rails/rails/blob/ba877bd088db8ec276d10e14700ee0b87dcdb747/activerecord/lib/active_record/railtie.rb#L212-L237

Reply:
Yes, it's a bit unfortunate, but some places where looking up `AR::Base.some_config` causing `AR::Base` autoloading and the `on_load` callback.

But now that they access `AR.some_config` they no longer trigger it.

I'm not particularly happy with this situation and am open to suggestion, because it's a bit unclear when configs are supposed to be accessible, and this reliance on autoload means referencing `AR::Base` early can cause the copy to happen earlier than expected.